docs/l10n.rst

```diff
@@ -9,17 +9,303 @@ Localization
+        page('about/contact', 'about/contact.html'),
+    ]
+
+To tell this page to use the Fluent framework for l10n you just need to tell it which file(s)
+to use:
+
+.. code-block:: python
+
+    urlpatterns = [
+        page('about', 'about.html', ftl_files='about'),
+        page('about/contact', 'about/contact.html', ftl_files=['about/contact', 'about']),
+    ]
+
+The system uses the first (or only) file in the list to determine which locales are active for that
+URL. You can pass a string or list of strings to the ``ftl_files`` argument. The files you specify
+can include the ``.ftl`` extension or not, and they will be combined with the list of default files
+which contain strings for global elements like navigation and footer. There will also be files for
+reusable widgets like the newsletter form, but those should always come last in the list.
+
+Using the class-based view
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Bedrock includes a generic class-based view (CBV) that sets up l10n for you. If you need to do anything fancier
+than just render the page, then you can use this:
+
+.. code-block:: python
+
+    from lib.l10n_utils import L10nTemplateView
+
+    class AboutView(L10nTemplateView):
+        template_name = 'about.html'
+        ftl_files = ['about']
+
+Using that CBV will do the right things for l10n, and then you can override other useful methods
+(e.g. ``get_context_data``) to do what you need. Also, if you do need to do anything fancy with
+the context, and you find that you need to dynamically set the fluent files list, you can easily do
+so by setting ``ftl_files`` in the context instead of the class attribute.
+
+.. code-block:: python
+
+    from lib.l10n_utils import L10nTemplateView
+
+    class AboutView(L10nTemplateView):
+        template_name = 'about.html'
+
+        def get_context_data(self, **kwargs):
+            ctx = super().get_context_data(**kwargs)
+            ftl_files = ['about']
+            if request.GET.get('fancy'):
+                ftl_files.append('fancy')
+
+            ctx['ftl_files'] = ftl_files
+            return ctx
+
+Using in a view function
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Lastly there's the good old function views. These should use ``l10n_utils.render`` directly to render
+the template with the context. You can use the ``ftl_files`` argument with this function as well.
+
+.. code-block:: python
+
+    from lib.l10n_utils import render
+
+    def about_view(request):
+        render(request, 'about.html', {'name': 'Duder'}, ftl_files='about')
+
+Fluent File Activation
+----------------------
+
+Fluent files are activated automatically when processed from the l10n team's repo
+into our own based on a couple of rules.
+
+1. If a fluent file has a group of required strings, all of those strings must be present in
+   the translation in order for it to be activated.
+2. A translation must contain a minimum percent of the string IDs from the English file to be activated.
```

Comment:
Where is the minimum percentage for activation set? (what is the default value?), and what should a developer do if they are creating a new page and want it to use a different percentage?